File: Python/217containsDuplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        # Use a set since it has an average time complexity of O(1) for insertion and element checks
        temp = set()
        for n in nums:
            # If the value is already in the set return true
            if n in temp:
                return True
            temp.add(n)
        # Exiting the loop signifies that no dupes were found
        return False

File: Java/217containsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        // Loop through each number in the array
        for (int n : nums) {
            // Return true if n is already in the set
            if (set.contains(n)) {
                return true;
            }
            // Add current n to the set
            set.add(n);
        }
        // If loop is exited then all values are unique
        return false;
    }
}